### PR TITLE
Custom atlas

### PR DESCRIPTION
**Pull request by akhanf**
- adds a `--atlas` cmdline option
- adds atlas wildcard to subfield outputs
- adds new magdeburg atlas (from ASHS data)
- bigbrain still the default
- ~no longer reliant on .mat file subfield definition, uses label.gii instead~ we use the unfolded volumetric space to store the atlases now

Note, I've marked this as a breaking change because it adds a `atlas` wildcard to the subfield output filenames.

To do:

- [x] need to iron out a minor issue moving from the .mat subfields to label.gii, I believe there is still a minor difference of 256x128 vs 254x126 dims, so the .mat subfields may actually contain more information at the boundaries than our 254x126 label.gii files.. @jordandekraker, thoughts? 
- [x] add to docs (incl link to create-atlas)

### CODE REVIEW

**Comment by jordandekraker** [comment]
It looks like the sizes (254x126 vs 256x128) isn't too bad of an issue since `linspace()` is being used to determine sample locations, so really we're just doing an unnecessary resampling here.

To simplify things a bit, one thing we could do is define labels with `space-unfolded.nii` + `label.gii` together. The nii could simply be warped to space-corobl, and the label.gii will remain the same in all cases anyway. This would also have the added benefit of allowing subfield boundaries to be slanted rather than perfectly vertical.

(if subfields boundaries were slanted, as many histologists draw them, then we might also want to consider having a different outer.label.gii from inner.label.gii, but that's a bit of a detail and might end up confusing later ROI-based analyses even more)

**Comment by jordandekraker** [comment]
Thinking more about it, a labelmap in the unfolded volumetric space (ie a .nii of the same size (or resampled to) the size of the refvol created in the line below) might be best since the `label.gii` will also change depending on `density`, but either way it can be sampled from an unfolded volumetric atlas. https://github.com/khanlab/hippunfold/blob/ff60d3a9012827cb96fe622e03b4fd11880d6ec7/hippunfold/workflow/rules/warps.smk#L33
This would also change https://github.com/khanlab/hippunfold-create-atlas a bit

I'll try out this method!

**Comment by akhanf** [comment]
I think having the labels defined in a single file (rather than multiple) is preferred, and then can have a rule to generate the other forms -- 

So I agree the unfolded volumetric space seems like an ideal place for it, it would be easy to sample this with unfolded surfaces (to create .label.gii files). Also does make it future proof for laminar-specific labels which is nice. And the ability to edit the labels in, say, itksnap, is attractive too (since editing label.gii directly in an interactive way isn't possible)..

**Comment by akhanf** [comment]
Cool,  had a look at your commit just now -- all looks good, except now for the volumetric subfield labelling, instead of using the native coords to label subfields (e.g. `label_subfields_from_vol_coords.py`) it is transforming the unfolded labels to native space, which uses the warp we got from scattered interpolation -- this is going to introduce some extra resampling error (e.g. since it is going native -> unfold -> native), instead of just labelling the native directly..

**Comment by jordandekraker** [comment]
Do you mean the added error is because I resampled the atlas to unfolded space? Or is there somewhere else in the code that we resample these subfields back to unfolded and to native again that I can't think of right now?
in effect, the atlas was always in unfolded space. The .mat was a 512x256 grid which we binned and downsampled. I think this should just as good. 

I'm just running a few test cases now too!

**Comment by akhanf** [comment]
No, that part is fine. I'm just referring to the one rule `label_subfields_from_unfolded_atlas` that previously was performing the native volumetric subfield labelling by using the native coords directly, whereas now you have modified it to use the Inverse warp to transform the subfields to native.. I haven't actually verified myself what differences this will have, but there likely will be some minor differences since e.g. if you transform the unfolded coords volume to native, you won't have exactly the same values as native coords (since we don't have perfect inverse-consistency in our warps).. 

But we should compare them to see whether things are noticeably different.. 
If so, the fix would just be to update `label_subfields_from_vol_coords.py` so it uses our label_nii instead of the label_gii..

**Comment by jordandekraker** [comment]
you're right, using the warp was definitely producing different results. I reverted it now and set up to use the .nii atlases.

**Comment by akhanf** [comment]
cool will have a look right now

File: hippunfold/config/snakebids.yml [hunk 1]
```diff
@@ -278,8 +286,14 @@ template_files:
     crop_refT1w: resources/CITI168/T1w_300umCoronalOblique_hemi-{hemi}.nii.gz
     Mask_crop: resources/CITI168/Mask_300umCoronalOblique_hemi-{hemi}.nii.gz
 
-
-
+atlas: bigbrain
+atlas_files:
+  bigbrain:
+    label_nii: resources/bigbrain/sub-bigbrain_hemi-{hemi}_label-hipp_desc-manualsubfields.nii.gz
```
Comment (akhanf) [inline hippunfold/config/snakebids.yml:292]: minor bids-naming comment here -- this file is missing a suffix..
suffix should be dseg here, I can make this change when I go through it now

**Review by akhanf — COMMENTED** [review]
looks great! I have a couple minor file naming changes to make, and will also try to add a documentation blurb if I get a chance, then I think we are good to merge

File: hippunfold/workflow/rules/gifti.smk [hunk 2]
```diff
@@ -549,60 +549,54 @@ rule calculate_thickness_from_surface:
...
+rule resample_atlas_to_refvol:
+    """this is just done in case the atlas has a different unfolded config than the current run"""
     input:
-        label=os.path.join(
-            workflow.basedir,
-            "..",
-            "resources",
-            "bigbrain",
-            "sub-bigbrain_hemi-{hemi}_subfields.label.gii",
+        atlas=os.path.join(
```
[the remark this answers is not in the document]
Comment (jordandekraker) [reply hippunfold/workflow/rules/gifti.smk:555]: good catch!

**Comment by akhanf** [comment]
ok battled a bit with linting because I had some version mixups (and I think discovered a snakefmt bug), but I've made those naming changes and also updated `label_subfields_from_vol_coords.py` to use the unfolded nifti.. 

I also ended up having it read in the IO coord too since it was easy to do, so now we do have the ability to define subfields in 3D in unfolded space. Could be useful if we define atlases with that information too (e.g. by warping subfields from native to unfolded). 

I'm just running a quick test to compare labels here vs labels from the master branch now..

**Comment by akhanf** [comment]
ok all looks good now, and I've updated the docs (outputs) to reflect the naming change, so will merge now.

**Pull request merged**